core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -148,6 +148,20 @@ case object TaskKilled extends TaskFailedReason {
 
 /**
  * :: DeveloperApi ::
+ * Task requested the driver to commit, but was denied.
+ */
+@DeveloperApi
+case class TaskCommitDenied(
+    jobID: Int,
+    splitID: Int,
```

Comment:
I actually think that partitionID might be clearer here.  If you look at how these values are set, they come from the `splitId` that's used to configure `SparkHadoopWriter`, which is a partition id in the range `0 to numPartitions`, whereas a taskId might be confused for a globally unique task identifier.